### ISSUE TITLE
Old links to library docs are broken

### ISSUE THREAD

**Issue by RalfJung** [coq/coq#18144]
Google search results are full of links like https://coq.inria.fr/library/Coq.Lists.List.html that currently no longer work. It'd be great to get proper redirects for these as right now, finding the right docs can be quite painful.

**Comment by Zimmi48** [comment]
Indeed. As [discussed on Zulip](https://coq.zulipchat.com/#narrow/stream/237656-Coq-devs-.26-plugin-devs/topic/Coq.20website.20migration), the issue is that GitHub Pages do not support HTTP redirects and it is not realistic to add HTML redirects for every sub-path. A JS-based solution in a 404 might be possible. Otherwise, the remaining options are to re-migrate to a static page provider that supports HTTP redirects (such as GitLab Pages), to leave these links broken, or to deploy the documentation in a library repository (similarly to what was done for refman).

Note that for the refman, the current state of things is that https://coq.inria.fr/distrib/.../index.html redirects to https://coq.inria.fr/doc/.../refman/index.html (but not for other sub-paths), including for `current` which redirects to the current version of the refman, but https://coq.inria.fr/refman uses a second deployment (second copy) of the refman hosted at https://github.com/coq/refman. The same could possibly be done for library (this is what I meant above as the last option).

**Comment by Zimmi48** [comment]
Let's migrate this issue to the proper repository.
